.changeset/two-pumpkins-wait.md

```diff
@@ -0,0 +1,49 @@
+---
+'@scale-codec/definition-compiler': major
+'@scale-codec/definition-runtime': major
+---
+
+**refactor**: use local opaque type pattern, remove `type-fest` dep (and `Opaque` re-export); workaround enum issues with `EnumBox` type; update generation, generate less code
+
+#### Opaque types
+
+Each generated codec type is **opaque** - it can be instantiated only through the codec factory or by explicit `as` conversion. It **was** implemented with `Opaque` type from `type-fest` library. Now it is implemented with the following technique:
```

Comment:
```suggestion
Each generated codec type is **opaque**, which means it can be instantiated only through the codec factory or by an explicit `as` conversion. It used to be implemented via `Opaque` type from `type-fest` library. Now it is implemented in the following way:
```